Mathlib/LinearAlgebra/Finsupp/Sym2.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2025 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Data.Finset.Sym
+import Mathlib.Data.Finsupp.Defs
+import Mathlib.Data.Sym.Sym2
+import Mathlib.Algebra.GroupWithZero.Defs
+
+/-!
+# Finitely Supported Commutative multiplication
+
+-/
+
+section
+
+variable {α R}
+
+variable [CommMonoidWithZero R]
+
+lemma Sym2.mem_support_sym2_of_mul_ne_zero {f : α →₀ R} (p : Sym2 α) (hp : mul (p.map f) ≠ 0) :
+    p ∈ f.support.sym2 := by
```

Comment:
Is this true as
```suggestion
lemma Sym2.sym2_support {f : α →₀ R} (p : Sym2 α) (hp : mul (p.map f) ≠ 0) :
    f.support.sym2 = map f ⁻¹' mul.support := by
```

Reply:
@YaelDillies 
```
lemma Sym2.sym2_support_eq_preimage_support_mul (f : α →₀ R) (p : Sym2 α) :
    f.support.sym2 = map f ⁻¹' mul.support := by
  ext p
  obtain ⟨a,b⟩ := p
  simp_all
  constructor
  · sorry
  · intro h
    aesop
```
Where the `sorry` is `¬f a = 0 ∧ ¬f b = 0 → ¬f a * f b = 0` which I don't think is true without further conditions on `R`?

Reply:
Yeah, you will need to assume `NoZeroDivisors`, so it should be a separate lemma. Can you make it a separate lemma?

Reply:
Btw `ext p; obtain ⟨a, b⟩ := p` is the same as `ext ⟨a,b⟩`

Reply:
Done, thanks.